webapp/src/js/components/containers/Gene.js

```diff
@@ -0,0 +1,210 @@
+          genomePositionTableQuery = SQL.WhereClause.encode(SQL.WhereClause.AND([
+            SQL.WhereClause.CompareFixed(this.config.tables[table].settings.chromosome, '=', geneData['chromid']),
+            SQL.WhereClause.CompareFixed(this.config.tables[table].settings.regionStart, '<=', parseInt(geneData['fstop'])),
+            SQL.WhereClause.CompareFixed(this.config.tables[table].settings.regionStop, '>=', parseInt(geneData['fstart']))
+          ]));
+        }
+
+        let genomePositionTableButton = (
+          <RaisedButton style={buttonStyle}
+                        key={table}
+                        label={'Show ' + table + ' in ' + geneData['fname']}
+                        primary={true}
+                        onClick={(e) => this.handleOpenTableTab(e, table, {query: genomePositionTableQuery})}
+                        icon={<Icon fixedWidth={true} name={this.config.tables[table].icon} inverse={true} />}
+          />
+        );
+        genomePositionTableButtons.push(genomePositionTableButton);
+      }
+
+    }
+
+    let externalGeneLinks = JSON.parse(this.config.settings.externalGeneLinks);
+    let externalGeneLinkButtons = [];
+    for (let i = 0, len = externalGeneLinks.length; i < len; i++) {
+      let externalGeneLinkButton = (
+        <RaisedButton style={buttonStyle}
+                      key={'externalGeneLinkButton_' + i}
+                      label={externalGeneLinks[i].Name}
+                      primary={true}
+                      onClick={(e) => this.handleOpenExternal(e, externalGeneLinks[i].Url.replace('{Id}', geneData['fid']))}
+                      icon={<Icon fixedWidth={true} name="external-link" inverse={true} />}
+        />
+      );
+      externalGeneLinkButtons.push(externalGeneLinkButton);
+    }
+
+    return (
+      <div>
+        <table className="table-col">
+          <tbody>
+            <tr><th className="table-col-header">Name: </th><td className="table-col-cell">{geneData['fname']}</td></tr>
+            <tr><th className="table-col-header">Alternatives: </th><td className="table-col-cell">{geneData['fnames'].split(',').join(', ')}</td></tr>
+            <tr><th className="table-col-header">Description: </th><td className="table-col-cell">{geneData['descr']}</td></tr>
+            <tr><th className="table-col-header">Position: </th><td className="table-col-cell">{geneData['chromid']}:{geneData['fstart']}-{geneData['fstop']}</td></tr>
+          </tbody>
+        </table>
+        <Loading status={loadStatus}/>
+        <div className="stack wrap">
+          <RaisedButton style={buttonStyle}
+                        label="Show in Genome Browser"
+                        primary={true}
+                        onClick={(e) => this.handleOpenGenomeBrowser(e, {chromosome: geneData['chromid'], start: parseInt(geneData['fstart']), end: parseInt(geneData['fstop'])})}
+                        icon={<Icon fixedWidth={true} name="bitmap:genomebrowser.png" inverse={true} />}
```

Comment:
This icon needs style={{filter: 'invert(100%)'}} to make the genome browser icon the right colour (or preferably we need to find a way to make all buttons do this)